Odata-docs/odatalib/v7/edm/iedmtargetpath-for-annotations-target.md

```diff
@@ -0,0 +1,463 @@
+            this.City = new EdmEntityType("NS", "City");
+            this.City.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            this.Model.AddElement(this.City);
+
+            this.NavUnderComplex = this.Address.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "City",
+                    Target = this.City,
+                    TargetMultiplicity = EdmMultiplicity.One,
+                });
+
+            this.NavUnderCustomer = this.Customer.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "Cities",
+                    Target = this.City,
+                    TargetMultiplicity = EdmMultiplicity.Many,
+                });
+
+            this.NameProperty = this.Customer.DeclaredProperties.Where(x => x.Name == "Name").FirstOrDefault();
+            this.AddressProperty = this.Customer.DeclaredProperties.Where(x => x.Name == "Address").FirstOrDefault();
+
+            this.Container = new EdmEntityContainer("NS", "Default");
+            this.EntitySet = new EdmEntitySet(this.Container, "Customers", this.Customer);
+            this.Singleton = new EdmSingleton(this.Container, "Me", this.Customer);
+            this.Container.AddElement(this.EntitySet);
+            this.Container.AddElement(this.Singleton);
+
+            this.Model.AddElement(this.Container);
+        }
+
+        public EdmModel Model { get; private set; }
+        public EdmEntityContainer Container { get; private set; }
+        public EdmEntitySet EntitySet { get; private set; }
+        public EdmSingleton Singleton { get; private set; }
+        public EdmEntityType Customer { get; private set; }
+        public EdmEntityType VipCustomer { get; private set; }
+        public EdmEntityType City { get; private set; }
+        public EdmComplexType Address { get; private set; }
+        public EdmComplexType DerivedAddress { get; private set; }
+        public IEdmProperty NameProperty { get; private set; }
+        public IEdmProperty AddressProperty { get; private set; }
+        public EdmNavigationProperty NavUnderComplex { get; private set; }
+        public EdmNavigationProperty NavUnderCustomer { get; private set; }
+    }
+}
+
+```
+
+## Write out-of-line annotations
+
+Replace the contents of _Program.cs_ file with the following code:
+
+```csharp
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Vocabularies;
+using Microsoft.OData.Edm;
+using System.Reflection;
+using System.ComponentModel;
+using Microsoft.OData.Edm.Validation;
+using System.Xml;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+using System.Xml.Linq;
+
+namespace AnnotationsTarget
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            SampleModel sampleModel = new SampleModel();
```

Comment:
For demo, maybe re-organize the 'main' body?
Create
1) one method to play building the annotation on target path during writing
3) one method to play consume the annotation on target path during reading.